### DISCUSSION TITLE
Saving R Baseplots As Objects & Plotting Those Objects

### DISCUSSION THREAD

**Discussion by Sreenath1986** [wlandau/targets-tutorial#16] [Q&A] [answered]
## Description

I am facing issues while trying to plot R base plots which are saved as targets objects. However, I am able to plot ggplot plots which are saved as targets objects,  without any issues. Please help me with this. Thank you.

### Below code goes in functions.R file
Define a function to create & return a R baseplot (boxplot in this case)
`
fn_baseplot <- function(dataset){
  t <- boxplot(mpg~cyl, data=mtcars)
  return(t)
}
`
Define a function to create & return a R ggplot (again boxplot)
`
fn_ggplot <- function(dataset){
  t <- boxplot(mpg~cyl, data=mtcars)
  return(t)
}
`
### Below code goes in _targets.R file
Create targets objects based on the above two functions
`
library(targets)
data(mtcars)
source("functions.R")
list(
        tar_target(tar_baseplot, fn_baseplot(mtcars)), 
        tar_target(tar_ggplot, fn_ggplot(mtcars))
)
`
### Verifying the targets objects
`
tar_load(tar_ggplot)
plot(tar_ggplot)
` 
 Prints the box plot
`
tar_load(tar_baseplot)
plot(tar_baseplot)
`
Below error message
_Error in xy.coords(x, y, xlabel, ylabel, log) : 
 'x' is a list, but does not have components 'x' and 'y'_
`

**Comment by wlandau** [comment, marked as answer]
A base plot is not an object in the R session, it lives in a graphics device of the current R session. You could save the plot to a file and track it with [`format = "file"`](https://books.ropensci.org/targets/data.html#external-files):

```r
# _targets.R file
library(targets)

save_plot <- function(path) {
  png(path)
  plot(mpg ~ cyl, data = mtcars)
  dev.off()
  path
}

list(
  tar_target(plot, save_plot("plot.png"), format = "file")
)
```

**Reply by Sreenath1986** [reply]
Thank you so much for the prompt response. It worked like a charm.
